src/tasks/netSdk/netSdkTaskUtils.ts

```diff
@@ -0,0 +1,133 @@
+        withNamedArg('-p:ContainerImageTag', NetSdkDefaultImageTag, { assignValue: true })
+    )();
+
+    const quotedArgs = Shell.getShellOrDefault().quote(args);
+    return quotedArgs.join(' ');
+}
+
+export async function getNetSdkRunCommand(context: DockerTaskExecutionContext): Promise<string> {
+    const client = await ext.runtimeManager.getClient();
+    const folderName = await quickPickWorkspaceFolder(context.actionContext, ErrMsgNoWorkplaceFolder);
+
+    const command = await client.runContainer({
+        detached: true,
+        publishAllPorts: true,
+        name: getDefaultContainerName(folderName.name, NetSdkDefaultImageTag),
+        environmentVariables: {},
+        removeOnExit: true,
+        imageRef: getDefaultImageName(folderName.name, NetSdkDefaultImageTag),
+        labels: defaultVsCodeLabels,
+        mounts: await getRemoteDebuggerMount(),
+        exposes: [8080], // hard coded for now since the default port is 8080
+        entrypoint: '/bin/sh'
+    });
+
+    const quotedArgs = Shell.getShellOrDefault().quote(command.args);
+    const commandLine = [client.commandName, ...quotedArgs].join(' ');
+    return commandLine;
+}
+
+async function inferProjPath(context: IActionContext, folder: WorkspaceFolder): Promise<string> {
+    const noProjectFileErrMessage = l10n.t('No .csproj file could be found.');
+    const item = await quickPickProjectFileItem(context, undefined, folder, noProjectFileErrMessage);
+    return item.absoluteFilePath;
+}
+
+/**
+ * This method normalizes the Docker OS type to match the .NET Core SDK conventions.
+ * {@link https://learn.microsoft.com/en-us/dotnet/core/rid-catalog}
+ */
+async function normalizeOsToRid(): Promise<'linux' | 'win'> {
+    const dockerOsType = await getDockerOSType();
+    return dockerOsType === 'windows' ? 'win' : 'linux';
+}
+
+/**
+ * This method normalizes the native architecture to match the .NET Core SDK conventions.
+ * {@link https://learn.microsoft.com/en-us/dotnet/core/rid-catalog}
+ */
+async function normalizeArchitectureToRid(): Promise<RidCpuArchitecture> {
+    const architecture = os.arch();
+    switch (architecture) {
+        case 'x32':
+        case 'ia32':
+            return 'x86';
+        default:
+            return architecture;
+    }
+}
+
+/**
+ * This methods returns the mount for the remote debugger ONLY as the SDK built container will have
+ * everything it needs to run the app already inside.
+ */
+async function getRemoteDebuggerMount(): Promise<RunContainerBindMount[] | undefined> {
+    const volumes: DockerContainerVolume[] = [];
+    const isWindows = await getDockerOSType() === 'windows';
+
+    const debuggerVolume: DockerContainerVolume = {
+        localPath: vsDbgInstallBasePath,
+        containerPath: isWindows ? 'C:\\remote_debugger' : '/remote_debugger',
+        permissions: 'ro'
+    };
+
+    addVolumeWithoutConflicts(volumes, debuggerVolume);
```

Comment:
Why not return a RunContainerBindMount directly? You're always starting with an empty set of volumes here, so addVolumeWithoutConflicts/getMounts aren't really doing anything here.

Reply:
Honestly, I don't love that we go back and forth between DockerContainerVolume and RunContainerBindMount here and in inferVolumes; I think I'd like it more if we normalized the volumes to RunContainerBindMount[] early and did all our volume logic with that type instead.